addons/web_editor/static/tests/html_field_tests.js

```diff
@@ -36,13 +124,199 @@ QUnit.module("WebEditor.HtmlField", ({ beforeEach }) => {
+                            }
+                            if (args.args[0] === "web_editor.snippets") {
+                                return SNIPPETS_TEMPLATE;
+                            }
+                        }
+                    },
+                },
+            }
+        });
+
+        await makeView({
+            type: "form",
+            resId: 1,
+            resModel: "note.note",
+            serverData,
+            arch: `
+                <form>
+                    <field name="body" widget="html" style="height: 100px"/>
+                </form>`,
+        });
+
+        const field = target.querySelector('.o_field_html[name="body"]');
+        const editable = field.querySelector("[contenteditable='true']");
+        await click(editable);
+
+        assert.strictEqual(
+            window.getComputedStyle(document.querySelector("#toolbar")).getPropertyValue("visibility"),
+            "hidden",
+            "should hide the toolbar");
+
+        // select the text
+        const pText = editable.querySelector("p").firstChild
+        Wysiwyg.setRange(pText, 1, pText, 10);
+        // text is selected
+
+        var range = Wysiwyg.getRange();
+
+        assert.strictEqual(range.sc, pText,
+            "should select the text");
+
+        await wait(1);
+        assert.strictEqual(
+            window.getComputedStyle(document.querySelector("#toolbar")).getPropertyValue("visibility"),
+            "visible",
+            "should show the toolbar");
+
+
+        assert.ok(!document.querySelector(".colorpicker-menu.show"),
+            "should hide the color picker");
+        await click(document.querySelector("#oe-fore-color"), undefined, true);
+        assert.ok(document.querySelector('.o_we_color_btn'),
+            "should display some button");
+        assert.ok(document.querySelector(".colorpicker-menu.show"),
+            "should show the color picker");
+        await click(document.querySelector('.o_we_color_btn[style="background-color:#F7C6CE;"]'), undefined, true);
+        await wait(100)
+        assert.ok(!document.querySelector(".colorpicker-menu.show"),
+            "should close the color picker");
+        assert.strictEqual(editable.innerHTML,
+            "<p>t<font style=\"background-color: rgb(247, 198, 206);\">oto toto </font>toto</p><p>tata</p>",
+            "should have rendered the field correctly");
+
+        var fontElement = document.querySelector('.note-editable font');
+        var rangeControl = {
+            sc: fontElement,
+            so: 0,
+            ec: fontElement,
+            eo: 1,
+        };
+        range = Wysiwyg.getRange();
+        assert.deepEqual(_.pick(range, 'sc', 'so', 'ec', 'eo'), rangeControl,
+            "should keep the selected the selected text after color change");
+
+        // select the text
+        const pText2 = [...document.querySelector('.note-editable p').childNodes][2];
+        Wysiwyg.setRange(fontElement.firstChild, 3, pText2, 2);
```

Comment:
This is different too.

Reply:
It's okay I think

Reply:
I'm not sure about this one because if you change the selection then the result should change too. If you change the selection but the result is the same as before (not counting the change of color) then something is definitely wrong. Event if the result would be properly adapted to match the new range (I don't think it currently is ?), again, I'm interested in understanding why you thought the previous range was bad and chose to change it.